<launch>
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="joy_topic" default="spot_teleop/joy" />
  <arg name="cmd_vel_topic" default="spot_teleop/cmd_vel" />
  
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="20" />

    <remap from="joy" to="$(arg joy_topic)" />
  </node>

  <node pkg="teleop_twist_joy" name="teleop_twist_joy" type="teleop_node">
    <remap from="joy" to="$(arg joy_topic)" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)" />

    <rosparam subst_value="True">
        axis_linear:
          x: 1
          y: 0
        scale_linear:
          x: 0.4
          y: 0.4
        scale_linear_turbo:
          x: 0.5
          y: 0.5
        axis_angular:
          yaw: 3
        scale_angular:
          yaw: 0.5
        scale_angular_turbo:
          yaw: 0.6
        enable_button: 4
        enable_turbo_button: 7
    </rosparam>
  </node>
</launch>
